scarlet/bbox.py

```diff
@@ -43,66 +62,199 @@ def from_bounds(bottom, top, left, right):
+                return False
+        return True
+
+    def slices_for(self, im_or_shape):
+        """Slices for `im_or_shape` to be limited to this bounding box.
+
+        Parameters
+        ----------
+        im_or_shape: array or tuple
+            Array or shpae of the array to be sliced
+
+        Returns
+        -------
+        If shape is 2D: `slice_y`, `slice_x`
+        If shape is 3: `slice(None)`, `slice_y`, `slice_x`
+        """
+        if hasattr(im_or_shape, 'shape'):
+            shape = im_or_shape.shape
+        else:
+            shape = im_or_shape
+        assert len(shape) in [2,3]
+
+        im_box = Box(shape)
+        overlap = self & im_box
+        zslice, yslice, xslice = slice(overlap.front, overlap.back), slice(overlap.bottom, overlap.top), slice(overlap.left, overlap.right)
+
+        if len(shape) == 2:
+            return yslice, xslice
+        else:
+            return zslice, yslice, xslice
+
+    def image_to_box(self, image, box=None):
+        """Excize box described by this bbox from image
+
+        Parameters
+        ----------
+        image: array
+            Full origin image
+        box: array
+            Excized destination image
+
+        Returns
+        -------
+        box: array
+        """
+        imbox = Box.from_image(image)
+
+        if box is None:
+            if len(image.shape) == 3:
+                box = np.zeros(self.shape)
+            else:
+                box = np.zeros(self.shape[1:])
+        boxbox = Box.from_image(box)
+
+        # imbox now in the frame of this bbox (i.e. of box)
+        imbox -= self.origin
+        overlap = imbox & boxbox
+        box[overlap.slices_for(box)] = image[self.slices_for(image)]
+        return box
+
+    def box_to_image(self, box, image):
+        """Insert `box` into `image` according to this bbox
+
+        Inverse operation to :func:`~scarlet.bbox.Box.image_to_box`.
+
+        Parameters
+        ----------
+        box: array
+            Excized box
+        image: array
+            Full image
+
+        Returns
+        -------
+        image: array
+        """
+        imbox = Box.from_image(image)
+        boxbox = Box.from_image(box)
+
+        # imbox now in the frame of this bbox (i.e. of box)
+        imbox -= self.origin
+        overlap = imbox & boxbox
+        image[self.slices_for(image)] = box[overlap.slices_for(box)]
+        return image
+
+    @property
+    def C(self):
+        """Number of channels in the model
+        """
+        return self.shape[0]
+
+    @property
+    def Ny(self):
+        """Number of pixel in the y-direction
+        """
+        return self.shape[1]
+
+    @property
+    def Nx(self):
+        """Number of pixels in the x-direction
         """
-        return self.width == 0 or self.height == 0
+        return self.shape[2]
 
     @property
-    def slices(self):
-        """Slices for this bounding box
+    def front(self):
```

Comment:
again, how are you ordering `back` in front of `front`?

Reply:
So I looked this up and it looks like some codes use back to front and others use front to back, so there doesn't seem to be a standard convention for the z-axis.

Personally I think of +z as back to front, since I think of the coordinate system as me standing at the origin looking in the positive direction in all three axes. My back is behind me and my front is in the positive direction, so I think "back to front" using that convention.

My guess is that your convention is walking in a house, where you start at the front and walk towards the back, which makes sense to me but is not the way that I orient things. So I guess it's fine to leave the way it is since it's arbitrary anyway.